src/projection-types.el

```diff
@@ -42,7 +42,8 @@
    :name 'haskell-cabal
    :predicate (defun projection-haskell-cabal-project-p ()
                 (and (file-expand-wildcards "?*.cabal")
-                     (not (file-exists-p "stack.yml"))))
+                     (not (file-exists-p "stack.yml"))
+                     (not (file-exists-p "stack.yaml"))))
```

Comment:
Can you reorder this in `projection-project-types` rather than disabling one when the other is supported or vice-versa. The predicate matching is useful to be able to expand all supported multi-compile targets for a project or to give you options of cycling the primary project type. 

I'm not too familiar with haskell, this was likely backported from projectile, but are you saying it never makes sense to run any cabal style commands in a project with a stack.yaml file?